syntax = "proto3";

// import "base.proto";

package communication.request;

// for Protocol.LOGIN
message TokenLogin {
    string token = 1;
    string name = 2;
}


message DirectoryVerification {
    string root_name = 1;
    map<string, string> contents = 2; // map<文件名, 文件内容md5值>
}

message FileDiff {
    string relative_path = 1;
    string content = 2;
    int32 status = 3;
}